shared/chat/conversation/messages/react-button/container.js

```diff
@@ -1,22 +1,45 @@
 // @flow
 import {compose, connect, setDisplayName, type TypedState} from '../../../../util/container'
-import * as MessageTypes from '../../../../constants/types/chat2/message'
+import * as Constants from '../../../../constants/chat2'
+import * as Types from '../../../../constants/types/chat2'
+import * as Chat2Gen from '../../../../actions/chat2-gen'
 import ReactButton from '.'
 
 export type OwnProps = {
+  conversationIDKey: Types.ConversationIDKey,
   emoji: string,
-  messageID: MessageTypes.MessageID,
+  ordinal: Types.Ordinal,
 }
 
 const mapStateToProps = (state: TypedState, ownProps: OwnProps) => {
-  // TODO
-  return {}
+  const me = state.config.username || ''
+  const message = Constants.getMessage(state, ownProps.conversationIDKey, ownProps.ordinal)
+  if (!message || message.type === 'placeholder' || message.type === 'deleted') {
```

Comment:
you should make a `const noEmoji = { active: false...` in the module and just return that in these cases

Reply:
good idea, done :+1: